vignettes/adtr.Rmd

```diff
@@ -0,0 +1,599 @@
+  filter(TUEVAL == "INVESTIGATOR")
+tr <- convert_blanks_to_na(tr) %>%
+  filter(TREVAL == "INVESTIGATOR" & TRGRPID == "TARGET" & TRTESTCD == "DIAMETER")
+rs <- convert_blanks_to_na(rs)
+```
+
+```{r echo=FALSE}
+# work-around for https://github.com/pharmaverse/admiral.test/issues/98
+# should be removed once the bug is fixed
+tr <- ungroup(tr)
+```
+
+```{r echo=FALSE}
+tr <- filter(tr, USUBJID %in% c("01-701-1015", "01-701-1023", "01-701-1028"))
+```
+
+At this step, it may be useful to join `ADSL` to your `TR` domain.  Only the
+`ADSL` variables used for derivations are selected at this step.  The rest of
+the relevant `ADSL` variables would be added later.
+
+```{r eval=TRUE}
+adsl_vars <- vars(RANDDT)
+tr <- derive_vars_merged(
+  tr,
+  dataset_add = adsl,
+  new_vars = adsl_vars,
+  by_vars = vars(STUDYID, USUBJID)
+)
+```
+
+## Merge TR with TU and derive new variables {#merge}
+
+Depending on your company specifications you may want to merge `TU` with `TR`.
+As an example, here we are keeping the tumor location, and deriving a new
+variable for the tumor group site. You may want to process other variables like
+`ADT`.
+
+```{r}
+tr <- derive_vars_merged(
+  tr,
+  dataset_add = rename(tu, TRLNKID = TULNKID),
+  new_vars = vars(TULOC),
+  by_vars = vars(STUDYID, USUBJID, TRLNKID)
+) %>% mutate(
+  TULOCGR1 = if_else(
+    TULOC == "LYMPH NODE",
+    "NODAL",
+    "NON-NODAL"
+  )
+)
+```
+
+```{r, echo=FALSE}
+dataset_vignette(tr, display_vars = vars(USUBJID, TRLNKID, TULOC, TULOCGR1))
+```  
+
+
+In addition, you could create any additional new variables that are required
+downstream when deriving the required parameters. For example, here we include
+lesion ID expected and lesion ID assessed. (any other examples?)
+
+```{r}
+tr <- mutate(
+  tr,
+  LSEXP = TRLNKID,
+  LSASS = if_else(!is.na(TRSTRESN), TRLNKID, NA_character_)
+)
+```
+
+## Pre-processing of input records {#input}
+
+The next step involves company-specific pre-processing of records for the
+required input to downstream parameter derivations.  See examples below.
+
+### Partial Date Imputation and Deriving `ADT`, `ADTF`, `AVISIT` etc
+
+If your data collection allows for partial dates, you could apply a
+company-specific imputation rule at this stage when deriving `ADT`. For this
+example, here we impute missing day to earliest possible date.
+
+```{r}
+tr <- derive_vars_dt(
+  tr,
+  dtc = TRDTC,
+  new_vars_prefix = "A",
+  highest_imputation = "D",
+  date_imputation = "first"
+) %>%
+  derive_vars_dy(
+    reference_date = RANDDT,
+    source_vars = vars(ADT)
+  )
+```
+
+
+```{r, echo=FALSE}
+dataset_vignette(tr, display_vars = vars(USUBJID, RANDDT, TRLNKID, TRDTC, ADT, ADTF, ADY))
+```  
+
+### Re-labelling unscheduled visits
+
+If you have data collected at unscheduled visits, which are not identified as
+unique in your SDTM or you have company specific rules.  Below is an example of
+how it could be re-labelled. <add more information once I see the template code>
```

Comment:
My comment is something similar. This chunk is not really handling unscheduled visit slotting.  So maybe just state that it needs to be considered, or add additional information indicating this can be quite complex

Reply:
I agree Roche derivation is complex.  If we say that grouping lesion diameter by visit is important for the calculations so we only count once, then we should either describe as you say needs to be considered. Maybe with a few details on how this could be approached instead of code?  How do you identify unique visits in the SDTM?

Example from BMS could describe at high level as text ?

Note:  There is a 1-1 relationship between [AVISIT] and [AVISITN]
BMS: Set to -1 if APHASE = "SCREENING".
if APHASE = "POST-BASELINE" then:

Derive temporary variable MAXADY as follows:

(1) For records with EVAL = "INVESTIGATOR": day (ADY) corresponding to the latest date recorded on CRF page ID (the CRF page identifier is recorded as TRSPID/TUSPID variable) in the corresponding period (APHASE).

(2) For records with EVAL = "INDEPENDENT ASSESSOR":

step 1. Find the Maximum date (MAXDATE):

use the latest RS.RSDTC when RSTESTCD = "OVRLRESP" and RSEVAL = "INDEPENDENT ASSESSOR" for corresponding evaluator (EVALID) and visit from RSSPID (If visit is "UNSCHEDULED" merge with " " and 5th substring of RSREFID using "-" as a separator). If BICR response is not available, use the latest ADLE.ADT per ADLE.CRELID for corresponding evaluator (EVALID), where ADLE.EVAL = "INDEPENDENT ASSESSOR".

step 2. Calculate MAXADY as:

MAXADY = MAXDATE - ADCORE.TRTSDT + 1.

If subject was randomized but not treated use ADCORE.RANDDT instead of ADCORE.TRTSDT.

Set to 1 if 2 <= MAXADY <= 11; why not 2 to 8??

set to 2 if 12 <= MAXADY <= 18;

set to 3 if 19 <= MAXADY <= 25;

continue for all other MAXADY values.


Note: AVISITN = x if 7 * x - 2 <= MAXADY <= 7 * x + 4

Reply:
as per our call. I have left this as it is, explained each company to update based on there requirements